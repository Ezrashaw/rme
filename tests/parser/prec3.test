(7 + 6) / 72351 - -func(var! + 1) * 4