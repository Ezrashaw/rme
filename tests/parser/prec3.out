formatted: (7 + 6) / 72351 - -func(var! + 1) * 4

Statement(Expression)@0..37
| BinaryOp@0..37
| | op: Sub@16..17
| | lhs: BinaryOp@0..15
| | | op: Div@8..9
| | | lhs: Paren@1..6
| | | | open: 0..1
| | | | close: 6..7
| | | | expr: BinaryOp@1..6
| | | | | op: Add@3..4
| | | | | lhs: Float(7.0)@1..2
| | | | | rhs: Float(6.0)@5..6
| | | rhs: Float(72351.0)@10..15
| | rhs: BinaryOp@18..37
| | | op: Mul@34..35
| | | lhs: UnaryOp@19..33
| | | | op: Negation@18..19
| | | | expr: FunctionCall("func"@19..23)@19..33
| | | | | open: 23..24
| | | | | close: 32..33
| | | | | arg #0 (comma): None
| | | | | arg #0: BinaryOp@24..32
| | | | | | op: Add@29..30
| | | | | | lhs: UnaryOp@24..27
| | | | | | | op: Factorial@27..28
| | | | | | | expr: Variable("var")@24..27
| | | | | | rhs: Float(1.0)@31..32
| | | rhs: Float(4.0)@36..37
